models/yiiModels/DataAnalysisAppSearch.php

```diff
@@ -0,0 +1,232 @@
+
+                    $descriptionText = $integratedFunctions[$functionName][self::FUNCTION_DESCRIPTION];
+                    $visualisationsInfo[$functionName][self::VIGNETTE_IMAGE] = "$descriptionPath/$functionName.png";
+                    $visualisationsInfo[$functionName][self::FUNCTION_HELP] = $descriptionText;
+                    $visualisationsInfo[$functionName][self::APP_SHORT_NAME] = explode("/", $app)[1] . "-" . $functionName;
+                    $visualisationsInfo[$functionName][self::R_PACKAGE_NAME] = $app;
+                    $url = "$applicationWebPath/index.html?accessToken=" . Yii::$app->session[WSConstants::ACCESS_TOKEN] . "&wsUrl=" . WS_PHIS_PATH;
+                    $visualisationsInfo[$functionName][self::APP_INDEX_HREF] = Url::to(['data-analysis/view', 'url' => $url, 'name' => explode("/", $app)[1]]);
+                }
+            }
+        }
+        return $visualisationsInfo;
+    }
+
+    /**
+     * Retreive information on demo app only
+     * @return array information on demo app
+     */
+    public function getAppDemoInformation() {
+        $visualisationsInfo = [];
+
+        if ($this->ocpuserver->status()) {
+            $app = self::DEFAULT_DEMO_APP;
+
+            $appConfiguration = $this->getAppConfiguration($app);
+            if (!empty($appConfiguration) 
+                    && isset($appConfiguration[self::AVAILABLE_FUNCTIONS][0]) 
+                    && isset($appConfiguration[self::INTEGRATED_FUNCTION])) {
+                // only one function
+                $functionName = $appConfiguration[self::AVAILABLE_FUNCTIONS][0];
+                $integratedFunctions = $appConfiguration[self::INTEGRATED_FUNCTION];
+
+                $applicationWebPath = $this->ocpuserver->getOpenCPUWebServerUrl() . "apps/" . $app . "/www";
+                $descriptionPath = $this->ocpuserver->getOpenCPUWebServerUrl() . "apps/" . $app . "/opensilex/description";
+
+                $descriptionText = $integratedFunctions[$functionName][self::FUNCTION_DESCRIPTION];
+                $visualisationsInfo[self::VIGNETTE_IMAGE] = "$descriptionPath/$functionName.png";
+                $visualisationsInfo[self::FUNCTION_HELP] = $descriptionText;
+                $visualisationsInfo[self::APP_SHORT_NAME] = explode("/", $app)[1] . "-" . $functionName;
+                $visualisationsInfo[self::R_PACKAGE_NAME] = $app;
+                $url = "$applicationWebPath/index.html?accessToken=" . Yii::$app->session[WSConstants::ACCESS_TOKEN] . "&wsUrl=" . WS_PHIS_PATH;
+                $visualisationsInfo[self::APP_INDEX_HREF] = Url::to(['data-analysis/view', 'url' => $url, 'name' => explode("/", $app)[1]]);
+            }
+        }
+        return $visualisationsInfo;
+    }
+
+    /**
+     * Return exported functionalities 
+     * @param string $applicationName 
+     * @return array functionalities
```

Comment:
Add @example of the format of the returned array

Reply:
Add explanation